### ISSUE TITLE
group import via api

### ISSUE THREAD

**Issue by apatedolos** [gophish/gophish#661]
Can anyone help with the issue I am having?

I have been trying to setup a script to group import, below is the code.

files = {'file':('filename.csv', open('filename.csv', 'rb'), 'application/vnd.ms-excel')}
headers = {'Content-Type': 'multipart/form-data;'}
url = "http://localhost:3333/api/import/group/"
resp = requests.post(url, params=api_key, files=files, headers=headers)

I have wire sharked the connection and I get a 301 redirect, followed by error parsing csv, see below.

POST /api/import/group/?api_key=redacted HTTP/1.1
Host: 192.168.0.38:3333
Connection: keep-alive
Accept-Encoding: gzip, deflate
Accept: */*
User-Agent: python-requests/2.12.4
Content-Type: multipart/form-data;
Content-Length: 228

--6b4930e6a0984996a8bac86170064cdf
Content-Disposition: form-data; name="file"; filename="filename.csv"
Content-Type: application/vnd.ms-excel

john,doe,director,john@doe.doe,JohnDoe

--6b4930e6a0984996a8bac86170064cdf--
HTTP/1.1 301 Moved Permanently
Location: /api/import/group?api_key=redacted
Vary: Accept-Encoding
Date: Tue, 20 Jun 2017 15:04:21 GMT
Content-Length: 0
Content-Type: text/plain; charset=utf-8

GET /api/import/group?api_key=redacted HTTP/1.1
Host: 192.168.0.38:3333
Connection: keep-alive
Accept-Encoding: gzip, deflate
Accept: */*
User-Agent: python-requests/2.12.4

HTTP/1.1 500 Internal Server Error
Content-Type: application/json
Vary: Accept-Encoding
Date: Tue, 20 Jun 2017 15:04:21 GMT
Content-Length: 72

{
  "message": "Error parsing CSV",
  "success": false,
  "data": null
}

Have I missed something simple?

**Comment by jordan-wright** [comment]
Try changing /group/ to /group in your script.

When a redirect happens (like Gophish moving your trailing slash to the right endpoint without the slash) your POST gets changed to a GET and your params get dropped, hence the error.

**Comment by apatedolos** [comment]
Great, thanks.

That has stopped the 301 redirect, however I still get a 500 error

HTTP/1.1 500 Internal Server Error
Content-Type: application/json
Vary: Accept-Encoding
Date: Wed, 21 Jun 2017 06:57:32 GMT
Content-Length: 72

{
  "message": "Error parsing CSV",
  "success": false,
  "data": null
}

**Comment by jordan-wright** [comment]
It looks like your CSV is missing the headers, check out our docs for an example: https://gophish.gitbooks.io/user-guide/content/campaign/importing_groups.html

**Comment by apatedolos** [comment]
hmm, I took the headers from here:

https://getgophish.com/documentation/api/#import-group-post

I assume by headers, you mean column headers?

**Comment by S0larflare** [comment]
I haven't tried importing through the api, but doing it thorugh the console (which should itself call the api) the headers are different.

I have it as "First Name", "Last Name", "Email", "Position". No underscores. Not sure why "Company" shows as a header either, I thought that one had been expunged a while ago!

**Comment by apatedolos** [comment]
Yeah, I can get the imports working fine through the console with the headers "First Name", "Last Name", "Email", "Position" as well, just not with the api, I was guessing that the "Company" header would formulate the Group name on the import. have tried many combinations but all fail to parse the CSV.

**Comment by S0larflare** [comment]
The "Company" header shouldn't be in there at all. Going by what you posted earlier:

> john,doe,director,john@doe.doe,JohnDoe

You have too many columns. It should simply be:

> john,doe,johndoe@company.com,director

**Comment by apatedolos** [comment]
yep, i tried that as well but still face the same issue.
{
"message": "Error parsing CSV",
"success": false,
"data": null
}
